lib/item_repo.rb

```diff
@@ -10,52 +10,45 @@ class ItemRepo
   include CreateElements
 
   def initialize(data, parent)
-    @items = create_elements(data).reduce({}) do |item_collection, item|
-      item_collection[item[:id].to_i] = Item.new(item)
-      # item_collection
+    @items = {}
+    create_elements(data).each do |row|
+      @items[row[:id]] = Item.new(row)
     end
     @parent = parent
   end
 
   def all
-    items.values
+    return @items
```

Comment:
It will need to be @items.values. This is because we want to return an array of all the objects, not the hash we are using.